db/controllers.js

```diff
@@ -12,15 +12,26 @@ exports.getTopics = (req, res) => {
       res.status(500).send({ msg: "server error" });
     });
 };
-
 exports.getEndpoints = (req, res) => {
   return fs.readFile("endpoints.json", `utf-8`, (err, data) => {
     if (err) {
       console.log(err);
       res.status(500).send("Couldn't read endpoints.json");
     } else {
-    const parsedEndpoints = JSON.parse(data)
-      res.send({ endpoints: parsedEndpoints});
+      const parsedEndpoints = JSON.parse(data);
+      res.send({ endpoints: parsedEndpoints });
     }
   });
 };
+exports.getArticleById = (req, res) => {
+  const id  = req.params.article_id
+  console.log(id);
+  selectArticleById(id)
+    .then((result) => {
+      res.status(200).send({result});
+  })
+    .catch((err) => {
+      if (err) console.log(err);
```

Comment:
There should not be any if statements or defined errors in this catch block. What you have written here is saying that no matter the error send back a 400. What if the article doesn't exist? That is not an invalid input. The error handling section of the notes has a very good example of what to do here.